snapshot_test.go

```diff
@@ -0,0 +1,139 @@
+		t.Fail()
+	}
+
+	return contents
+}
+
+func TestSnapshot(t *testing.T) {
+	snapshotter := cupaloy.New(
+		cupaloy.SnapshotSubdirectory("snapshots"),
+		cupaloy.SnapshotFileExtension(".go"),
+	)
+
+	type testCase struct {
+		name               string
+		outputFile         string
+		inputInterface     string
+		expectedOutputFile string
+	}
+
+	testCases := []testCase{
+		{
+			name:               "package reference",
+			outputFile:         "./tests",
+			inputInterface:     "github.com/gojuno/minimock/v3.Tester",
+			expectedOutputFile: "tests/tester_mock_test.go",
+		},
+		{
+			name:               "relative reference",
+			inputInterface:     "./tests.Formatter",
+			outputFile:         "./tests/formatter_mock.go",
+			expectedOutputFile: "./tests/formatter_mock.go",
+		},
+		{
+			name:               "generics with any used as param and return type",
+			inputInterface:     "./tests.genericInout",
+			outputFile:         "./tests/generic_inout.go",
+			expectedOutputFile: "./tests/generic_inout.go",
+		},
+		{
+			name:               "generics with any used as return type",
+			inputInterface:     "./tests.genericOut",
+			outputFile:         "./tests/generic_out.go",
+			expectedOutputFile: "./tests/generic_out.go",
+		},
+		{
+			name:               "generics with any used as param type",
+			inputInterface:     "./tests.genericIn",
+			outputFile:         "./tests/generic_in.go",
+			expectedOutputFile: "./tests/generic_in.go",
+		},
+		{
+			name:               "generics with specific type used as a generic constraint",
+			inputInterface:     "./tests.genericSpecific",
+			outputFile:         "./tests/generic_specific.go",
+			expectedOutputFile: "./tests/generic_specific.go",
+		},
+		{
+			name:               "generics with simple union used as a generic constraint",
+			inputInterface:     "./tests.genericSimpleUnion",
+			outputFile:         "./tests/generic_simple_union.go",
+			expectedOutputFile: "./tests/generic_simple_union.go",
+		},
+		{
+			name:               "generics with complex union used as a generic constraint",
+			inputInterface:     "./tests.genericComplexUnion",
+			outputFile:         "./tests/generic_complex_union.go",
+			expectedOutputFile: "./tests/generic_complex_union.go",
+		},
+		{
+			name:               "generics with complex inline union used as a generic constraint",
+			inputInterface:     "./tests.genericInlineUnion",
+			outputFile:         "./tests/generic_inline_union.go",
+			expectedOutputFile: "./tests/generic_inline_union.go",
+		},
+		{
+			name:               "generics with complex inline union with many types",
+			inputInterface:     "./tests.genericInlineUnionWithManyTypes",
+			outputFile:         "./tests/generic_inline_with_many_options.go",
+			expectedOutputFile: "./tests/generic_inline_with_many_options.go",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
According to the Go documentation (`go help testflag` and `go help build`), t.Parallel() execution is automatically limited by GOMAXPROCS (as the default value). So, in our case (single package), the binary will already be limited by GOMAXPROCS

> 	-parallel n
	    Allow parallel execution of test functions that call t.Parallel, and
	    fuzz targets that call t.Parallel when running the seed corpus.
	    The value of this flag is the maximum number of tests to run
	    simultaneously.
	    While fuzzing, the value of this flag is the maximum number of
	    subprocesses that may call the fuzz function simultaneously, regardless of
	    whether T.Parallel is called.
	    By default, -parallel is set to the value of GOMAXPROCS.
	    Setting -parallel to values higher than GOMAXPROCS may cause degraded
	    performance due to CPU contention, especially when fuzzing.
	    Note that -parallel only applies within a single test binary.
	    The 'go test' command may run tests for different packages
	    in parallel as well, according to the setting of the -p flag
	    (see 'go help build').
	    
I can definitely change this to sequential if you're still having doubts about the CPU load during the execution, this will only change the execution time, not the result

Reply:
Ok, let's leave it as is if it's already bound by GOMAXPROCS, yep my concern was CPU load especially if the number of tests grows.